### ISSUE TITLE
Misleading unnecessary braces warning

### ISSUE THREAD

**Issue by bugadani** [rust-lang/rust#104141]
Const generics, combined with the `cfg!` macro gives an "unnecessary braces" warning. However, removing the braces results in a compilation failure.

Example:

https://rust.godbolt.org/z/Kv8Eq5hzn

I expect no warning to be emitted for the following code:

```rust
pub fn foo<const BAR: bool> () {}

pub fn main() {
    // warning: unnecessary braces around const expression
    foo::<{cfg!(feature = "foo")}>();
}
```
